gui/elements/qt/_dimensions.py

```diff
@@ -0,0 +1,85 @@
+from PyQt5.QtCore import Qt, QSize
+from PyQt5.QtWidgets import QWidget, QSlider, QGridLayout
+
+class QtDimensions(QWidget):
+    def __init__(self, viewer):
+        super().__init__()
+        self.viewer = viewer
+        self.sliders = []
+
+        layout = QGridLayout()
+        layout.setContentsMargins(0, 0, 0, 0)
+        self.setLayout(layout)
+        self.setFixedHeight(0)
+
+    def _axis_to_row(self, axis):
+        dims = len(self.viewer.dimensions.indices)
+        message = f'axis {axis} out of bounds for {dims} dims'
+
+        if axis < 0:
+            axis = dims - axis
+            if axis < 0:
+                raise IndexError(message)
+        elif axis >= dims:
+            raise IndexError(message)
+
+        if axis < 2:
+            raise ValueError('cannot convert y/x-axes to rows')
+
+        return axis - 2
```

Comment:
Again, this whole function points to something wrong with the design. Am I the only one who sees this?

Reply:
@jni definitely – this is what #86 (a WIP) is meant to (eventually) address.